src/autopas/containers/linkedCells/traversals/C04CellHandler.h

```diff
@@ -0,0 +1,437 @@
+   * Partial sums of sizes of combined buffers to determine start and end quickly
+   */
+  std::vector<std::vector<std::vector<unsigned long>>> _combinationSlicesOffsets;
+
+  /**
+   * Writes buffer content back to cell.
+   * @param cells
+   * @param baseIndex Index of base cell.
+   * @param combinationSlice
+   * @param combinationSlicesOffsets
+   * @param bufferSlice Index of slice in combinationSlice (source)
+   * @param cellSlice Index of slice in _baseOffsets (destination)
+   */
+  void writeBufferIntoCell(std::vector<ParticleCell> &cells, const unsigned long baseIndex,
+                           std::vector<ParticleCell> &combinationSlice,
+                           std::vector<std::vector<unsigned long>> &combinationSlicesOffsets,
+                           const unsigned long bufferSlice, const unsigned long cellSlice);
+
+  /**
+   * Writes cell content into buffer.
+   * @param cells
+   * @param baseIndex Index of base cell.
+   * @param combinationSlice
+   * @param combinationSlicesOffsets
+   * @param bufferSlice Index of slice in combinationSlice (destination)
+   * @param cellSlice Index of slice in _baseOffsets (source)
+   */
+  void writeCellIntoBuffer(const std::vector<ParticleCell> &cells, const unsigned long baseIndex,
+                           std::vector<ParticleCell> &combinationSlice,
+                           std::vector<std::vector<unsigned long>> &combinationSlicesOffsets,
+                           const unsigned int bufferSlice, const unsigned int cellSlice);
+
+  /**
+   * Creates offset intervals (stored in offset) from _cellPairOffsets
+   * @param _cellPairOffsets Source for interval creation.
+   */
+  void setupIntervals(std::vector<std::vector<std::pair<unsigned long, unsigned long>>> &_cellPairOffsets);
+};
+
+template <class ParticleCell, class PairwiseFunctor, DataLayoutOption DataLayout, bool useNewton3>
+inline void C04CellHandler<ParticleCell, PairwiseFunctor, DataLayout, useNewton3>::processBaseCell(
+    std::vector<ParticleCell> &cells, unsigned long x, unsigned long y, unsigned long z) {
+  const unsigned long baseIndex = utils::ThreeDimensionalMapping::threeToOneD(x, y, z, _cellsPerDimension);
+
+  // get all information for current thread
+  const auto threadID = autopas_get_thread_num();
+  auto &currentSlice = _currentSlices[threadID];
+  auto &combinationSlice = _combinationSlices[threadID];
+  auto &combinationSlicesOffsets = _combinationSlicesOffsets[threadID];
+
+  const size_t numSlices = _baseOffsets.size();
+
+  // First cell needs to initialize whole buffer
+  if (x == 0ul) {
+    currentSlice = 0ul;
+    for (unsigned int slice = 0ul; slice < numSlices; slice++) {
+      writeCellIntoBuffer(cells, baseIndex, combinationSlice, combinationSlicesOffsets, slice, slice);
+    }
+  } else {
+    writeCellIntoBuffer(cells, baseIndex, combinationSlice, combinationSlicesOffsets, currentSlice, numSlices - 1ul);
+
+    ++currentSlice %= numSlices;
+  }
+
+  // compute interactions
```

Comment:
I really hope you make a good visualization of this in your thesis :D
Maybe we then can include a sketch of it in the doxygen doc.